bigflow/scaffold/infra.py

```diff
@@ -0,0 +1,159 @@
+"""This module is not part of the public API. No backwards compatibility guaranteed."""
+
+import typing as tp
+import logging
+import functools
+from bigflow.commons import run_process
+
+logger = logging.getLogger(__name__)
+
+InfrastructureDestroyer = tp.Callable[[], None]
+InfrastructureCreator = tp.Callable[[], None]
+BIGFLOW_INFRA_PREFIX = 'bigflow'
+
+
+def secure_destroy(infrastructure_part_name: str) -> tp.Callable[[InfrastructureDestroyer], InfrastructureDestroyer]:
+    def secure_destroy_decorator(destroyer: InfrastructureDestroyer) -> InfrastructureDestroyer:
+        @functools.wraps(destroyer)
+        def secured_destroyer() -> None:
+            logger.info(f'Destroying infrastructure: {infrastructure_part_name}')
+            try:
+                destroyer()
+            except Exception as e:
+                logger.exception(f"Couldn't destroy infrastructure: {infrastructure_part_name}", exc_info=e)
+
+        return secured_destroyer
+
+    return secure_destroy_decorator
+
+
+def _create_cloud_router(
+        gcp_project_id: str,
+        bigflow_project_name: str,
+        environment_name: str,
+        region: str) -> tp.Tuple[str, InfrastructureDestroyer, InfrastructureCreator]:
+    router_name = f'{BIGFLOW_INFRA_PREFIX}-router-{bigflow_project_name}-{environment_name}-{region}'
+
+    @secure_destroy(f'{router_name} (Google Cloud router)')
+    def destroy() -> None:
+        logger.info(run_process([
```

Comment:
BTW `run_process` may dump all its output to console with `verbose=True` (it is a default).

Reply:
fixup: `run_process` always dumps its output to `logger`. Flag `verbose` switches between `info` and `debug` log levels.

Reply:
done